lib/trento.pm

```diff
@@ -826,6 +828,43 @@ sub cluster_wait_status {
     die "Timeout waiting for the change" if !$done;
 }
 
+=head3 cluster_wait_status_by_regex
+
+Remotly run 'SAPHanaSR-showAttr' in a loop on $host, wait output that match using regular expression
+=cut
+
+sub cluster_wait_status_by_regex {
+    my ($host, $timeout, $regular_expression, $text_to_find) = @_;
+    $timeout //= bmwqemu::scale_timeout(300);
+    my $_monitor_start_time = time();
+    my $done;
+    while ((time() - $_monitor_start_time <= $timeout) && (!$done)) {
+        sleep 30;
+        my $prov = get_required_var('PUBLIC_CLOUD_PROVIDER');
+        my $show_attr = qesap_ansible_script_output(
+            cmd => 'SAPHanaSR-showAttr',
+            provider => $prov,
+            host => $host,
+            root => 1);
+
+        my %status = ();
+        for my $line (split("\n", $show_attr)) {
+            $status{$1} = $line if ($line =~ $regular_expression);
+        }
+        my $hash_string = Dumper(\%status);
```

Comment:
I don't understand the logic behind this block between lines 851 & 854. Basically you parse the lines from the showAttr output so that only the ones that match the `$regular_expression` go into `%status`, but then you don't do anything with the structure created in `%status` and instead turn it into a string with `Dumper` where you look for the other expression.

Why not check directly in the showAttr output if you're going to check a string with a regexp?

From my other comments, I can see that the expressions in use are:

```
m/global.+true/, "true"
m/miky\s+SFAIL/, "SFAIL"
m/global.+false/, "false"
m/miky\s+SOK/, "SOK"
```

Going back to my 2 regular expressions question, seems to me the whole thing can be accomplished by either matching directly into `$show_attr` with a more precise multi-line regular expression, or - better yet - dropping `$text_to_find` and simply doing `$done = 1` when you see a line that matches to `$regular_expression`. No need to create the `%status` hash, or to then turn it into a string using `Dumper`.

What you're doing would be akin to doing the following in python:
```
import re
status = {}
for line in show_attr.split("\n"):
    result = re.match(regular_expression, line)
    if result:
        status[result.group(1)] = line
if re.match(text_to_find, str(status)):
    done = 1
else
    done = 0
```
Notice that nothing is done with `status` except turning it into a string? It's defined as a dict, but it could easily be a list, a concatenated string or some other data type. Same issue is present in the Perl code.

Reply:
thx for the comment, refactored